### PR TITLE
Adds Voice Changer and randomization to CARL's inventory

### PR DESCRIPTION
**Pull request by Zonespace27**
[FEATURE]
## About the PR 
The standard Syndicate items that CARL sells, (Radio Jammer, Stealth Storage, and the new Voice Changer) are now randomized, allowing for round-to-round variance of CARL's inventory.

In addition, Voice Changers are now in the list of possible items for CARL to sell, valued at 4000 credits.

The odds of items being sold are as follows:

20%: Radio Jammer
30%: Stealth Storage
30%: Voice Changer
10%: All of the three
10%: None
(This is all handled in one rand(), so, for example, there can't be only Voice Changers and Stealth Storages.)

## Why's this needed? 
Guaranteed Syndicate items can no longer be obtained by rushing the Listening Post, so that traitors don't have a consistent way of not having to buy certain items. In addition, this prevents crewmembers from doing the same thing and obtaining traitor items round after round.

Old text:
~~Voice Changers are a fun tool that can supplement many gimmicks, and are a very useful traitor item. Generally, if you're in the listening post, you have an Agent Card, which would work well with a Voice Changer to go with it. In addition, I don't see this majorly de-balancing anything, as the powerful signal jammer, a 3 TC item, is sold at the listening post as well.~~

## Changelog

```changelog
(u)Zonespace
(*)C.A.R.L., the Syndicate merchant, now sells Voice Changers.
(*)Part of C.A.R.L.'s inventory is now random.
```

### CODE REVIEW

**Comment by penny64** [comment]
Consider: there's no cordon, and therefore the but number of ways for players to break in with no agent card are completely limitless.

Some of the ways to break in seem intentional
Is it okay if everyone has access to a voice changer?

**Comment by Zonespace27** [comment]
> Consider: there's no cordon, and the number of ways for players to break in with no agent card are completely limitless.
> 
> Some of the ways to break in seem intentional
> Is it okay if everyone has access to a voice changer?

Voice Changers are already available from merchants that come to the station, and I haven't seen that abused to any degree. While I do get the difference between the possibility of a merchant selling them, versus CARL always selling them, I feel that the benefits can outweigh the negatives, especially when you consider things like some gimmicks, where your face is covered, but you aren't able to talk without revealing your identity.

**Comment by penny64** [comment]
@Zonespace27 nah sounds like you thought about it to me 👍

**Comment by Carbadox** [comment]
I think doing this devalues the point of buying the voice changer from a traitor PDA. Im aware that you can already buy a stealth storage from CARL but I don't believe more Syndicate items should be added to it.

Travelling merchants are chance based, you can't consistently get their stock. CARL doesnt move and he's also accesible through what I consider exploits. Not a fan of this change

**Comment by PrimeNumb** [comment]
Suggestion: have CARL sell one, and only one, item out of the set of items { Voice changer, Signal jammer, Stealth storage }. Which item is sold at any given round is picked randomly (possibly with different weights) when the commodity list is generated. You could also make it so that he has a chance to sell none of the above.

Why is that better?
A) Similar to random merchants, you would no longer be able to consistently get these traitor-only items from CARL. They would still be meaningful purchases for traitors.
B) Crew members won't be able to get these items consistently from CARL either, for the same reasons.

**Comment by Zonespace27** [comment]
> Suggestion: have CARL sell one, and only one, item out of the set of items { Voice changer, Signal jammer, Stealth storage }. Which item is sold at any given round is picked randomly (possibly with different weights) when the commodity list is generated. You could also make it so that he has a chance to sell none of the above.
> 
> Why is that better?
> A) Similar to random merchants, you would no longer be able to consistently get these traitor-only items from CARL. They would still be meaningful purchases for traitors.
> B) Crew members won't be able to get these items consistently from CARL either, for the same reasons.

CARL's new sell list looks like this:
20%: Radio Jammer
30%: Stealth Storage
30%: Voice Changer
10%: All of the three
10%: None

File: code/modules/economy/trader.dm [hunk 1]
```diff
@@ -733,6 +733,7 @@
 	trader_area = "/area/turret_protected/robot_trade_outpost"
 	var/productset = 0 // 0 is robots and salvage, 1 is podparts and drugs, 2 is produce. 3 is syndicate junk, 4 is medical stuff
 	var/illegal = 0 // maybe trading with illegal bots could flag the user's criminal record for smuggling
+	var/carlsell
```
Comment (Tarmunora) [inline code/modules/economy/trader.dm:736]:
```suggestion
```

File: code/modules/economy/trader.dm [hunk 2]
```diff
@@ -787,10 +788,19 @@
 				src.goods_buy += new /datum/commodity/produce/special/glowfruit(src)
 
 			if(3) // syndicate bot
+				carlsell = rand(1,10)
```
Comment (Tarmunora) [inline code/modules/economy/trader.dm:791]:
```suggestion
				var/carlsell = rand(1,10)
```

**Pull request merged**